src/app/pages/Homescreen/Homescreen.tsx

```diff
@@ -1,27 +1,49 @@
 import React from 'react';
+import { PopularMovie } from '../../../types';
 import NavigationGenre from '../../components/NavigationGenre/NavigationGenre';
 import Rating from '../../components/Rating/Rating';
 import MovieCard from '../../components/MovieCard/MovieCard';
 import styles from './Homescreen.module.css';
 import NavBar from '../../components/NavBar/NavBar';
+import useFetch from '../../hooks/useFetch';
 
 const categoriesArray = ['Movie', 'Adventure', 'Comedy', 'Family'];
 
 function Homescreen(): JSX.Element {
+  const { data: popularMovies, isLoading, errorMessage } = useFetch<
+    PopularMovie[]
+  >('api/movies/popular');
```

Comment:
This could produce an error if the path or your URL is not `/`. 
Imagine you are on `/dashboard`, which creates this API URL: `/dashboard/api/movies/popular`.
On `/` it is correct (`/api/movies/popular`).

```suggestion
  >('/api/movies/popular');
```

This makes sure, that it's always  `/api/movies/popular`.

Reply:
Thanks i changed it.